ballsdex/packages/trade/menu.py

```diff
@@ -240,6 +251,12 @@ async def cancel(self, reason: str = "The trade has been cancelled."):
         for countryball in self.trader1.proposal + self.trader2.proposal:
             await countryball.unlock()
 
+        self.trader1.player.coins += self.trader1.coins
+        self.trader2.player.coins += self.trader2.coins
+
+        await self.trader1.player.save()
+        await self.trader2.player.save()
+
```

Comment:
```suggestion
            await countryball.unlock()
```
this shouldn't be there, otherwise a user can just add 500 coins, cancel the trade and get the coins from his own proposal back